master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java

```diff
@@ -246,62 +245,27 @@ public void writeMetaInfoToFile(File file) throws IOException, RuntimeException
     ObjectOutputStream out =
         new ObjectOutputStream(new BufferedOutputStream(new FileOutputStream(file)));
 
-    out.writeLong(estimatedPartitionSize);
-    // write registeredShuffle
-    writeSetMetaToFile(registeredShuffle, out);
-
-    // write hostnameSet
-    writeSetMetaToFile(hostnameSet, out);
-
-    // write blacklist
-    writeSetMetaToFile(blacklist, out);
-
-    // write workerLost events
-    writeSetMetaToFile(workerLostEvents, out);
-
-    // write application heartbeat time
-    out.writeInt(appHeartbeatTime.size());
-    appHeartbeatTime.forEach(
-        (app, time) -> {
-          try {
-            out.writeObject(app);
-            out.writeLong(time);
-          } catch (IOException e) {
-            throw new RuntimeException(e);
-          }
-        });
-
-    // write workerInfo
-    out.writeInt(workers.size());
-    workers.forEach(
-        workerInfo -> {
-          try {
-            out.writeObject(workerInfo);
-          } catch (IOException e) {
-            throw new RuntimeException(e);
-          }
-        });
-
-    out.writeLong(partitionTotalWritten.sum());
-    out.writeLong(partitionTotalFileCount.sum());
-    out.writeObject(appDiskUsageMetric.snapShots());
-    out.writeObject(appDiskUsageMetric.currentSnapShot());
-
+    SnapshotMetaInfo snapshot =
+        new SnapshotMetaInfo(
+            estimatedPartitionSize,
+            registeredShuffle,
+            hostnameSet,
+            blacklist,
+            workerLostEvents,
+            appHeartbeatTime,
+            workers,
+            partitionTotalWritten.sum(),
+            partitionTotalFileCount.sum(),
+            appDiskUsageMetric.snapShots(),
+            appDiskUsageMetric.currentSnapShot().get());
+
+    byte[] snapshotBytes = PbSerDeUtils.toPbSnapshotMetaInfo(snapshot).toByteArray();
+
+    out.writeInt(snapshotBytes.length);
```

Comment:
what's the benefit of appending the length to the file?

if we make it as a pure pb binary file, we may can leverage the `protoc` to analyze the file content conveniently.

Reply:
Like current?